src/timepicker/timepicker.js

```diff
@@ -112,16 +119,20 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
   };
 
   state = {steps: [], value: null};
+  static contextType: React.Context<DateUtilsContextT> = DateUtilsContext;
+  context: DateUtilsContextT;
 
   componentDidMount() {
+    const {utils} = this.context;
     const steps = this.buildSteps();
-    if (this.props.value instanceof Date) {
+
+    if (utils.isValid(this.props.value) && this.props.value) {
```

Comment:
Using `instanceof` was specific enough that the type could be refined to Date inside of the if statement, this was the only way I could find to keep it working the same way